AppCenterDataStorage/AppCenterDataStorage/MSDataStore.h

```diff
@@ -0,0 +1,132 @@
+ */
++ (void)setTokenExchangeUrl:(NSString *)tokenExchangeUrl;
+
+/**
+ * Read a document.
+ * The document type (T) must be JSON deserializable.
+ */
++ (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
++ (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)readOptions
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
+/**
+ * List of documents. Use optional signature to configure page size.
+ * The document type (T) must be JSON deserializable.
+ */
++ (void)listWithPartition:(NSString *)partition
+             documentType:(Class)documentType
+        completionHandler:(MSPaginatedDocumentsCompletionHandler)completionHandler;
+
++ (void)listWithPartition:(NSString *)partition
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)readOptions
+        completionHandler:(MSPaginatedDocumentsCompletionHandler)completionHandler;
+
+/**
+ * Create a document.
+ * The document instance (T) must be JSON serializable.
+ */
++ (void)createWithPartition:(NSString *)partition
+                 documentId:(NSString *)documentId
+                   document:(T)document
+          completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
++ (void)createWithPartition:(NSString *)partition
+                 documentId:(NSString *)documentId
+                   document:(T)document
+               writeOptions:(MSWriteOptions *)writeOptions
+          completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
+/**
+ * Replace a document.
+ * The document instance (T) must be JSON serializable.
+ */
++ (void)replaceWithPartition:(NSString *)partition
+                  documentId:(NSString *)documentId
+                    document:(T)document
+           completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
++ (void)replaceWithPartition:(NSString *)partition
+                  documentId:(NSString *)documentId
+                    document:(T)document
+                writeOptions:(MSWriteOptions *)writeOptions
+           completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler;
+
+/**
+ * Delete a document.
+ */
++ (void)deleteDocumentWithPartition:(NSString *)partition
+                         documentId:(NSString *)documentId
+                  completionHandler:(MSDataStoreErrorCompletionHandler)completionHandler;
+
++ (void)deleteDocumentWithPartition:(NSString *)partition
+                         documentId:(NSString *)documentId
+                       writeOptions:(MSWriteOptions *)writeOptions
+                  completionHandler:(MSDataStoreErrorCompletionHandler)completionHandler;
+
+//
```

Comment:
remove these for now since, we are not implementing yet

Reply:
Removed